pmlc/target/x86/stencil_tpp_binary.cc

```diff
@@ -92,9 +108,15 @@ class StencilImpl : public pxa::StencilBase {
 
     auto source1 = cast<pxa::PxaLoadOp>(load1.getDefiningOp()).memref();
     auto source2 = cast<pxa::PxaLoadOp>(load2.getDefiningOp()).memref();
-
+    auto outputSource = cast<pxa::PxaReduceOp>(reduce.getDefiningOp()).memref();
     if (isLocallyDefined(op, source1) || isLocallyDefined(op, source2))
       return;
+    auto outputType = outputSource.getType().cast<MemRefType>();
+    auto input1Type = source1.getType().cast<MemRefType>();
+    auto input2Type = source2.getType().cast<MemRefType>();
+
+    if (!shapeCompatible(outputType, input1Type, input2Type))
+      return;
```

Comment:
It seems like this could just be:

```
if (outputType.getShape() != input1Type.getShape() || outputType.getShape() != input2Type.getShape())
  return;
```

Reply:
Ok, I'll make this change, thanks